src/Socket_p.h

```diff
@@ -116,6 +117,10 @@ namespace Arcus
         std::deque<MessagePtr> receiveQueue;
         std::mutex receiveQueueMutex;
 
+        std::mutex receiveQueueMutexBlock;
+        bool message_ready = false;
+        std::condition_variable socket_block_condition_variable;
```

Comment:
I think this variable has a bad name. It really guards the `message_ready` field, so its name should reflect that.

Reply:
I propose : message_received_condition_variable
:S